internal/glance/glance.go

```diff
@@ -439,6 +442,23 @@ func (a *application) server() (func() error, func() error) {
 	mux.HandleFunc("GET /{$}", a.handlePageRequest)
 	mux.HandleFunc("GET /{page}", a.handlePageRequest)
 
+	var absDataPath string
+	if a.Config.Server.DataPath == "" {
+		a.Config.Server.DataPath = "./data"
+	}
+	absDataPath, _ = filepath.Abs(a.Config.Server.DataPath)
+
+	fs := afero.NewOsFs()
+	todosDataPath := path.Join(absDataPath, "todos")
+	exist, _ := afero.DirExists(fs, todosDataPath)
+	if !exist {
+		fs.MkdirAll(todosDataPath, os.ModePerm)
+	}
+
+	todoWidget := &todoWidget{storage: afero.NewBasePathFs(fs, todosDataPath)}
```

Comment:
You're creating a second reference to the todo widget, in addition to the widgets.go file. I'd suggest moving that logic into the widgets.go file.

Defining the widgets data folder should imho be dependent on the widgets name and not be freely picked here.